---
title: "QC of FASTQ files"
author: "Deepak Tanwar"
date: "<b>Created on:</b> 2021-09-17 <br> <b>Updated on:</b> `r Sys.Date()`"
output:
  rmdformats::readthedown:
    highlight: tango
    self_contained: true
    thumbnails: true
    lightbox: true
    gallery: false
    keep_md: no
    number_sections: no
    fig_width: 8
    fig_height: 8
    fig_caption: true
    df_print: paged
    code_folding: hide
  fontsize: 12pt
  geometry: margin=1in
  documentclass: article
# bibliography: references.bib
link-citations: yes
---


# Library
```{r qc-report-1, warning=FALSE, message=FALSE}
library(Rfastp)
library(qckitfastq)
library(plotly)
library(plyr)
```

```{r qc-report-2, echo=FALSE,include = FALSE, warning=FALSE, message=FALSE}
# dependencies attaching...
plot_ly()
```

# Data
```{r qc-report-3, warning=FALSE, message=FALSE}
folder <- "../01_data"
files <- list.files(path = folder, pattern = ".gz", full.names = T)
names(files) <- gsub(pattern = "\\.fastq.gz", replacement = "", x = basename(files))
```

# `Rfastp`

## Analysis
```{r qc-report-4, warning=FALSE, message=FALSE}
qc_SE <- function(file, outdir, name, ad1) {
  res <- rfastp(
    read1 = file,
    outputFastq = paste(outdir, name, sep = "/"),
    thread = 4,
    cutMeanQual = 20,
    cutFrontMeanQual = 20,
    cutTailMeanQual = 20,
    cutLowQualTail = T,
    cutLowQualFront = T,
    qualityFiltering = T,
    maxNfilter = 5,
    qualityFilterPhred = 20,
    qualityFilterPercent = 40,
    lengthFiltering = T,
    minReadLength = 15,
    maxIndexMismatch = 1,
    adapterTrimming = TRUE,
    adapterSequenceRead1 = ad1
  )
  # file.remove(paste0(outdir, "/", name, ".html"))
  return(res)
}

res <- parallel::mclapply(setNames(names(files), names(files)), function(x) {
  qc_SE(file = files[[x]], outdir = "output/", name = x, ad1 = "TGGAATTCTCGGGTGCCAAGG")
}, mc.preschedule = F, mc.cores = 16)

json_files <- list.files(path = "output", pattern = "json", full.names = T)
names(json_files) <- gsub(pattern = "\\.json", replacement = "", x = basename(json_files))

trim_files <- list.files(path = "output", pattern = "gz", full.names = T)
names(trim_files) <- gsub(pattern = "\\_R1.fastq.gz", replacement = "", x = basename(trim_files))
```


## Reading and getting information from JSON file

### General
```{r qc-report-5, results='asis', warning=FALSE, message=FALSE}
#' Make general information data frame from
#'
#' @param json A JSON file with QC information
#'
#' @return A data frame with FASTQ file information
#' @export
#'
#' @examples
make_general_df <- function(json) {
  js <- rjson::fromJSON(file = json)

  # pkg <- as.character(packageVersion("Rfastp"))
  seq <- js$read1_before_filtering$total_cycles
  seq_type <- ifelse(test = any(grepl(pattern = "read2", x = names(js))), yes = "paired-end", no = "single-end")
  cycle <- paste0(seq_type, " (", seq, " cycles)")

  mean_len_bf <- js$summary$before_filtering$read1_mean_length
  mean_len_af <- js$summary$after_filtering$read1_mean_length
  dup_rate <- ifelse(test = seq_type == "paired-end",
    yes = paste0(round(x = (js$duplication$rate * 100), digits = 2), "%"),
    no = paste0(
      round(x = (js$duplication$rate * 100), digits = 2),
      "% (may be overestimated since this is SE data)"
    )
  )

  df <- data.frame(
    test = c(
      # "Rfastp version",
      "Sequencing",
      "Mean length before filtering",
      "Mean length after filtering",
      "Duplication rate"
    ),
    value = c(
      # pkg,
      cycle,
      mean_len_bf,
      mean_len_af,
      dup_rate
    )
  )
  return(df)
}

parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  knitr::kable(make_general_df(json = json_files[[x]]), col.names = NULL)
}, mc.preschedule = F, mc.cores = 4)
```

### Before QC
```{r qc-report-6, results='asis', warning=FALSE, message=FALSE}
make_before_filt_df <- function(json) {
  js <- rjson::fromJSON(file = json)
  tot_reads <- js$read1_before_filtering$total_reads
  tot_bases <- js$read1_before_filtering$total_bases
  q_20_bases <- js$read1_before_filtering$q20_bases
  q_20_rate <- js$summary$before_filtering$q20_rate * 100
  q_30_bases <- js$read1_before_filtering$q30_bases
  q_30_rate <- js$summary$before_filtering$q30_rate * 100
  gc <- paste0(round(x = js$summary$before_filtering$gc_content * 100, digits = 2), "%")

  df <- data.frame(
    test = c(
      "Total reads",
      "Total bases",
      "Q20 bases",
      "Q30 bases",
      "GC content"
    ),
    value = c(
      tot_reads,
      tot_bases,
      paste0(q_20_bases, " (", q_20_rate, "%)"),
      paste0(q_30_bases, " (", q_30_rate, "%)"),
      gc
    )
  )
  return(df)
}


parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  knitr::kable(make_before_filt_df(json = json_files[[x]]), col.names = NULL)
}, mc.preschedule = F, mc.cores = 4)
```

### After QC
```{r qc-report-7, results='asis', warning=FALSE, message=FALSE}
make_after_filt_df <- function(json) {
  js <- rjson::fromJSON(file = json)
  tot_reads <- js$read1_after_filtering$total_reads
  tot_bases <- js$read1_after_filtering$total_bases
  q_20_bases <- js$read1_after_filtering$q20_bases
  q_20_rate <- js$summary$after_filtering$q20_rate * 100
  q_30_bases <- js$read1_after_filtering$q30_bases
  q_30_rate <- js$summary$after_filtering$q30_rate * 100
  gc <- paste0(round(x = js$summary$after_filtering$gc_content * 100, digits = 2), "%")

  df <- data.frame(
    test = c(
      "Total reads",
      "Total bases",
      "Q20 bases",
      "Q30 bases",
      "GC content"
    ),
    value = c(
      tot_reads,
      tot_bases,
      paste0(q_20_bases, " (", q_20_rate, "%)"),
      paste0(q_30_bases, " (", q_30_rate, "%)"),
      gc
    )
  )
  return(df)
}


parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  knitr::kable(make_after_filt_df(json = json_files[[x]]), col.names = NULL)
}, mc.preschedule = F, mc.cores = 4)
```


### Filtering result
```{r qc-report-8, results='asis', warning=FALSE, message=FALSE}
make_filt_df <- function(json) {
  js <- rjson::fromJSON(file = json)

  reads <- js$summary$before_filtering$total_reads
  read_pass <- js$filtering_result$passed_filter_reads
  read_pass_pct <- round((read_pass / reads) * 100, 2)
  read_low_Q <- js$filtering_result$low_quality_reads
  read_low_Q_pct <- round((read_low_Q / reads) * 100, 2)
  read_with_N <- js$filtering_result$too_many_N_reads
  read_with_N_pct <- round((read_with_N / reads) * 100, 2)
  read_short <- js$filtering_result$too_short_reads
  read_short_pct <- round((read_short / reads) * 100, 2)
  read_long <- js$filtering_result$too_long_reads
  read_long_pct <- round((read_pass / reads) * 100, 2)

  df <- data.frame(
    test = c(
      "Reads passed filters",
      "Reads with low quality",
      "Reads with too many N",
      "Reads too short",
      "Reads too long"
    ),
    value = c(
      paste0(read_pass, " (", read_pass_pct, "%)"),
      paste0(read_low_Q, " (", read_low_Q_pct, "%)"),
      paste0(read_with_N, " (", read_with_N_pct, "%)"),
      paste0(read_short, " (", read_short_pct, "%)"),
      paste0(read_long, " (", read_long_pct, "%)")
    )
  )
  return(df)
}


parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  knitr::kable(make_filt_df(json = json_files[[x]]), col.names = NULL)
}, mc.preschedule = F, mc.cores = 4)
```


## Duplicated reads plot
```{r qc-report-9, results='asis', warning=FALSE, message=FALSE}
make_dup_plot <- function(json) {
  js <- rjson::fromJSON(file = json)

  bars <- js$duplication$histogram
  gc <- js$duplication$mean_gc
  xaxis <- 1:length(bars)

  data <- data.frame(
    Duplicate = round(bars / sum(bars) * 100, 2),
    GC = round(gc * 100, 2),
    x = xaxis
  )
  data$GC[data$GC == 0] <- ""

  p <- plot_ly(data, x = ~x) %>%
    add_trace(
      y = ~Duplicate,
      type = "bar",
      name = "Read percent",
      hoverinfo = "text+x",
      hovertext = paste(
        "<b>", data$Duplicate, "</b>"
      )
    ) %>%
    add_lines(
      y = ~GC,
      name = "Mean GC ratio (%)",
      line = list(color = "red", width = 3),
      hoverinfo = "text+x",
      hovertext = paste(
        "<b>", data$GC, "</b>"
      )
    ) %>%
    layout(
      title = paste0("Duplication rate: ", round(js$duplication$rate * 100, 2), "%"),
      xaxis = list(title = "Duplication level"),
      yaxis = list(title = "Reads percent & GC ratio (%)"),
      hovermode = "x closest"
    )
  print(htmltools::tagList(p))
}

parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_dup_plot(json = json_files[[x]])
}, mc.preschedule = F, mc.cores = 4)
```


## Read quality plot
```{r qc-report-10, warning=FALSE, message=FALSE, results='asis'}
make_read_qual_plot <- function(json, which = "before") {
  js <- rjson::fromJSON(file = json)

  qual <- NULL
  if (which == "before") {
    qual <- js$read1_before_filtering$quality_curves
  } else {
    qual <- js$read1_after_filtering$quality_curves
  }

  data <- data.frame(
    pos = 1:length(qual$mean),
    Aa = qual$A,
    Tt = qual$T,
    Gg = qual$G,
    Cc = qual$C,
    Mm = qual$mean
  )

  data$Q20 <- 20
  data$Q28 <- 8
  data$Q50 <- 18

  cols <- rcartocolor::carto_pal(n = 10, name = "Safe")[c(2, 1, 3, 4, 10)]

  p <- plot_ly(data, x = ~pos) %>%
    add_trace(
      y = ~Q20, name = "Not good", fillcolor = "rgba(255,0,0, 0.2)", type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Q28, name = "OK", fillcolor = "rgba(255,255,0,0.2)", type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Q50, name = "Good", fillcolor = "rgba(0,128,0,0.2)", type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_lines(
      y = ~Aa,
      name = "A",
      line = list(color = cols[1], width = 2),
      hoverinfo = "text",
      hovertext = paste("<b>", data$Aa, "</b>")
    ) %>%
    add_lines(
      y = ~Tt,
      name = "T",
      line = list(color = cols[2], width = 2),
      hoverinfo = "text",
      hovertext = paste("<b>", data$Tt, "</b>")
    ) %>%
    add_lines(
      y = ~Gg,
      name = "G",
      line = list(color = cols[3], width = 2),
      hoverinfo = "text",
      hovertext = paste("<b>", data$Gg, "</b>")
    ) %>%
    add_lines(
      y = ~Cc,
      name = "C",
      line = list(color = cols[4], width = 2),
      hoverinfo = "text",
      hovertext = paste("<b>", data$Cc, "</b>")
    ) %>%
    add_lines(
      y = ~Mm,
      name = "Mean",
      line = list(color = "black", width = 3, dash = "dot"),
      hoverinfo = "text",
      hovertext = paste("<b>", data$Mm, "</b>")
    ) %>%
    layout(
      title = "Per base sequence quality",
      xaxis = list(title = "Position in read (bp)"),
      yaxis = list(title = "Quality score"),
      hovermode = "x closest"
    )

  print(htmltools::tagList(p))
}
```

### Before QC
```{r qc-report-11, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_read_qual_plot(json = json_files[[x]], which = "before")
}, mc.preschedule = F, mc.cores = 4)
```

### After QC
```{r qc-report-12, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_read_qual_plot(json = json_files[[x]], which = "after")
}, mc.preschedule = F, mc.cores = 4)
```



## Base content line plot
```{r qc-report-13, warning=FALSE, message=FALSE, results='asis'}
make_base_ratio_line_plot <- function(json, which = "before") {
  js <- rjson::fromJSON(file = json)

  qual <- NULL
  if (which == "before") {
    qual <- js$read1_before_filtering$content_curves
  } else {
    qual <- js$read1_after_filtering$content_curves
  }

  data <- data.frame(
    pos = 1:length(qual$A),
    Aa = qual$A * 100,
    Tt = qual$T * 100,
    Gg = qual$G * 100,
    Cc = qual$C * 100,
    GC = qual$GC * 100,
    Nn = qual$N * 100
  )

  cols <- rcartocolor::carto_pal(n = 11, name = "Safe")[c(2, 1, 3, 4, 10)]

  p <- plot_ly(data, x = ~pos) %>%
    add_lines(
      y = ~Aa,
      name = "A",
      line = list(color = cols[1], width = 2),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$Aa, "</b>")
    ) %>%
    add_lines(
      y = ~Tt,
      name = "T",
      line = list(color = cols[2], width = 2),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$Tt, "</b>")
    ) %>%
    add_lines(
      y = ~Gg,
      name = "G",
      line = list(color = cols[3], width = 2),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$Gg, "</b>")
    ) %>%
    add_lines(
      y = ~Cc,
      name = "C",
      line = list(color = cols[4], width = 2),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$Cc, "</b>")
    ) %>%
    add_lines(
      y = ~GC,
      name = "GC",
      line = list(color = cols[5], width = 3, dash = "dot"),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$GC, "</b>")
    ) %>%
    add_lines(
      y = ~Nn,
      name = "N",
      line = list(color = "black", width = 3),
      hoverinfo = "text+x",
      hovertext = paste("<b>", data$Nn, "</b>")
    ) %>%
    layout(
      title = "Per base sequence content",
      xaxis = list(title = "Position in read (bp)/ Cycle"),
      yaxis = list(title = "Percentage (%)"),
      hovermode = "x closest"
    )
  print(htmltools::tagList(p))
}
```

### Before QC
```{r qc-report-14, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_base_ratio_line_plot(json = json_files[[x]], which = "before")
}, mc.preschedule = F, mc.cores = 4)
```

### After QC
```{r qc-report-15, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_base_ratio_line_plot(json = json_files[[x]], which = "after")
}, mc.preschedule = F, mc.cores = 4)
```


## Base content ratio proportion plot
```{r qc-report-16, warning=FALSE, message=FALSE, results='asis'}
make_base_ratio_prop_plot <- function(json, which = "before") {
  js <- rjson::fromJSON(file = json)

  qual <- NULL
  if (which == "before") {
    qual <- js$read1_before_filtering$content_curves
  } else {
    qual <- js$read1_after_filtering$content_curves
  }

  data <- data.frame(
    pos = 1:length(qual$A),
    Aa = qual$A * 100,
    Tt = qual$T * 100,
    Gg = qual$G * 100,
    Cc = qual$C * 100,
    GC = qual$GC * 100,
    Nn = qual$N * 100
  )

  cols <- rcartocolor::carto_pal(n = 11, name = "Safe")[c(2, 1, 3, 4, 10)]

  p <- plot_ly(data, x = ~pos) %>%
    add_trace(
      y = ~Nn, name = "N", fillcolor = "black", type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Cc, name = "C", fillcolor = cols[4], type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Gg, name = "G", fillcolor = cols[3], type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Tt, name = "T", fillcolor = cols[2], type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_trace(
      y = ~Aa, name = "A", fillcolor = cols[1], type = "scatter",
      mode = "none", stackgroup = "one", hoverinfo = "skip"
    ) %>%
    add_lines(
      y = ~GC,
      name = "GC",
      line = list(color = cols[5], width = 3, dash = "dot"),
      hoverinfo = "text",
      hovertext = paste("<b>", data$GC, "</b>")
    ) %>%
    layout(
      title = "Per base sequence content",
      xaxis = list(title = "Position in read (bp)/ Cycle"),
      yaxis = list(title = "Percentage (%)"),
      hovermode = "x closest"
    )
  print(htmltools::tagList(p))
}
```

### Before QC
```{r qc-report-17, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_base_ratio_prop_plot(json = json_files[[x]], which = "before")
}, mc.preschedule = F, mc.cores = 4)
```

### After QC
```{r qc-report-18, warning=FALSE, message=FALSE, results='asis'}
parallel::mclapply(setNames(names(json_files), names(json_files)), function(x) {
  make_base_ratio_prop_plot(json = json_files[[x]], which = "after")
}, mc.preschedule = F, mc.cores = 4)
```


# `qckitfastq`

## Analysis
```{r qc-report-19, warning=FALSE, message=FALSE}
run_qc <- function(fq) {
  out <- list()

  infile <- fq
  fseq <- seqTools::fastqq(infile)

  ## Read length
  read_len <- read_length(fseq)
  # plot_read_length(read_len)

  ## Per base sequence quality
  bs <- per_base_quality(infile)
  # plot_per_base_quality(bs)

  ## Per read quality
  prq <- per_read_quality(infile)
  # plot_per_read_quality(prq)

  ## GC content
  gc_df <- GC_content(infile)
  # plot_GC_content(gc_df)

  ## Nucleotide read content
  rc <- read_content(fseq)
  # plot_read_content(rc)

  ## Kmer count
  km <- kmer_count(infile, k = 6)

  ## Overrep reads
  # overrep_reads <- overrep_reads(infile)
  # plot_overrep_reads(overrep_reads)

  ## Overrep kmer
  overkm <- overrep_kmer(infile, 7)
  # plot_overrep_kmer(overkm)

  ## Adapter content
  if (.Platform$OS.type != "windows") {
    ac_sorted <- adapter_content(infile)
    # plot_adapter_content(ac_sorted)
  } else {
    ac_sorted <- "adapter_content not available for Windows; skipping"
    print("adapter_content not available for Windows; skipping")
  }

  out <- list(
    read_length = read_len,
    per_base_quality = bs,
    per_read_quality = prq,
    GC_content = gc_df,
    read_content = rc,
    kmer_count = km,
    overrep_kmer = overkm,
    overrep_reads = overrep_reads,
    adapter_content = ac_sorted
  )
  return(out)
}


res1 <- parallel::mclapply(setNames(names(files), names(files)), function(x) {
  run_qc(fq = files[[x]])
}, mc.preschedule = F, mc.cores = 4)

res2 <- parallel::mclapply(setNames(names(trim_files), names(trim_files)), function(x) {
  run_qc(fq = trim_files[[x]])
}, mc.preschedule = F, mc.cores = 4)
```


## Read length distribution
```{r qc-report-20, warning=FALSE, message=FALSE, results='asis'}
#' Read length distribution plot
#'
#' @param before_trim A dataframe with 2 columns: read_length and num_reads
#' @param after_trim A dataframe with 2 columns: read_length and num_reads
#'
#' @return An interactive barplot
#' @export
#'
#' @examples
read_length_plot <- function(before_trim, after_trim) {
  if (!colnames(before_trim) %in% c("read_length", "num_reads") |
    !colnames(after_trim) %in% c("read_length", "num_reads")) {
    message("Please check the columns of input")
  }
  raw <- before_trim
  trim <- after_trim
  raw$Group <- "Raw"
  trim$Group <- "Trimmed"

  p1 <- plot_ly(hoverinfo = "text", type = "bar", textposition = "auto") %>%
    add_trace(
      data = raw,
      legendgroup = ~Group,
      x = ~read_length, y = ~num_reads,
      name = ~Group,
      # text = ~`%Reads`,
      marker = list(color = "red"),
      hovertext = paste(
        "<b>Length:</b> ", raw$read_length,
        "<br><b>nReads:</b> ", raw$num_reads
        # , "<br><b>%Reads:</b> ", t$`%Reads`
      )
    ) %>%
    layout(
      title = "Histogram of read length distribution",
      xaxis = list(title = "Reads length"),
      yaxis = list(title = "No. of reads"),
      hovermode = "x closest"
    )

  p2 <- plot_ly(hoverinfo = "text", type = "bar", textposition = "auto") %>%
    add_trace(
      data = trim,
      legendgroup = ~Group,
      x = ~read_length, y = ~num_reads,
      name = ~Group,
      # text = ~`%Reads`,
      marker = list(color = "green"),
      hovertext = paste(
        "<b>Length:</b> ", trim$read_length,
        "<br><b>nReads:</b> ", trim$num_reads
        # , "<br><b>%Reads:</b> ", t$`%Reads`
      )
    ) %>%
    layout(
      title = "Histogram of read length distribution",
      xaxis = list(title = "Reads length"),
      yaxis = list(title = "No. of reads"),
      hovermode = "x closest"
    )
  p <- subplot(p1, p2, shareX = T, shareY = T)
  print(htmltools::tagList(p))
}

parallel::mclapply(setNames(names(res1), names(res1)), function(x) {
  read_length_plot(before_trim = res1[[x]]$read_length, after_trim = res2[[x]]$read_length)
}, mc.preschedule = F, mc.cores = 4)
```


# References
```{r qc-report-21, warning=FALSE, message=FALSE, eval=FALSE}
report::cite_packages(session = sessionInfo())
```

# SessionInfo
```{r qc-report-22, warning=FALSE, message=FALSE}
devtools::session_info() %>%
  details::details()
```
